modules/c++/mem/tests/ScratchVisualization.cpp

```diff
@@ -0,0 +1,477 @@
+            currentOperations.push_back(Operation("put", segmentName, numElements));
+            ++testIter;
+            ++bufferName;
+        }
+        else if (testIter == 1)
+        {
+            const std::string segmentName = std::string(1, bufferName) +
+                    str::toString(mIteration);
+            scratch.put<sys::ubyte>(segmentName, numElements, 1, 1);
+            currentOperations.push_back(Operation("put", segmentName, numElements));
+            ++testIter;
+            ++bufferName;
+        }
+        else if (testIter == 2)
+        {
+            const std::string segmentName = std::string(1, bufferName - 1) +
+                    str::toString(mIteration);
+            scratch.release(segmentName);
+            currentOperations.push_back(Operation("release", segmentName, numElements));
+            ++testIter;
+        }
+        else if (testIter == 3)
+        {
+            const std::string segmentName = std::string(1, bufferName) +
+                    str::toString(mIteration);
+            scratch.put<sys::ubyte>(segmentName, numElements, 1, 1);
+            currentOperations.push_back(Operation("put", segmentName, numElements));
+            ++testIter;
+            ++bufferName;
+        }
+        else if (testIter == 4)
+        {
+            const std::string segmentName = std::string(1, bufferName - 3) +
+                    str::toString(mIteration);
+            scratch.release(segmentName);
+            currentOperations.push_back(Operation("release", segmentName, numElements));
+            ++testIter;
+        }
+        else if (testIter == 5)
+        {
+            const std::string segmentName = std::string(1, bufferName) +
+                    str::toString(mIteration);
+            scratch.put<sys::ubyte>(segmentName, numElements, 1, 1);
+            currentOperations.push_back(Operation("put", segmentName, numElements));
+            ++testIter;
+            ++bufferName;
+        }
+        else if (testIter == 6)
+        {
+            const std::string segmentName = std::string(1, bufferName - 2) +
+                    str::toString(mIteration);
+            scratch.release(segmentName);
+            currentOperations.push_back(Operation("release", segmentName, numElements));
+            testIter = 0;
+        }
+    }
+
+private:
+    std::vector<std::string> mColors;
+    std::vector<std::string> mReleasedColors;
+    std::vector<Operation> mPrevOperations;
+    sys::ubyte* mStartPtr;
+    size_t mIteration;
+
+    std::ofstream& mHTMLFile;
+    std::ofstream& mCSSFile;
+};
+
+int main(int argc, char** argv)
+{
+    cli::ArgumentParser parser;
+
+    parser.setDescription("Software to visualize scratch memory test cases in HTML/CSS");
+    parser.addArgument("--test", "Select which test case to run", cli::STORE, "test")->setDefault("random");
+    parser.addArgument("--numBytes", "Determines how much memory is allocated for the buffer",
```

Comment:
Instead of --numBytes, can we just use the actual number of bytes from the ScratchMemory class? This would also show us that it allocates the size of buffer correctly.